datashader/layout.py

```diff
@@ -90,22 +90,22 @@ def __call__(self, nodes, edges, **params):
         return df
 
 
-def _extract_points_from_nodes(nodes):
+def _extract_points_from_nodes(nodes, params, dtype=None):
     if 'x' in nodes.columns and 'y' in nodes.columns:
         points = np.asarray(nodes[['x', 'y']])
     else:
-        points = np.asarray(np.random.random((len(nodes), 2)))
+        points = np.asarray(np.random.random((len(nodes), params.dim)), dtype=dtype)
     return points
 
 
-def _convert_graph_to_sparse_matrix(nodes, edges, dtype=None, format='csr'):
+def _convert_graph_to_sparse_matrix(nodes, edges, params, dtype=None, format='csr'):
     nlen = len(nodes)
     if 'id' in nodes:
         index = dict(zip(nodes['id'].values, range(nlen)))
     else:
         index = dict(zip(nodes.index.values, range(nlen)))
 
-    if 'weight' not in edges:
+    if params.ignore_weights or 'weight' not in edges:
         edges = edges.copy()
         edges['weight'] = np.ones(len(edges))
```

Comment:
This seems like a very expensive step, and doesn't seem feasible as an approach in general, but I guess we can merge and come back to that later.